src/Microsoft.Extensions.Configuration.DockerSecrets/DockerSecretsConfigurationBuilderExtensions.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using Microsoft.Extensions.Configuration.DockerSecrets;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Extension methods for registering <see cref="DockerSecretsConfigurationProvider"/> with <see cref="IConfigurationBuilder"/>.
+    /// </summary>
+    public static class DockerSecretsConfigurationBuilderExtensions
```

Comment:
@glennc can you take a look at the sugar methods and see if we need more (perhaps ones like optional)